conjure-java-core/src/main/java/com/palantir/conjure/java/util/SafetyUtils.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.util;
+
+import com.palantir.conjure.spec.AliasDefinition;
+import com.palantir.conjure.spec.ArgumentDefinition;
+import com.palantir.conjure.spec.ExternalReference;
+import com.palantir.conjure.spec.FieldDefinition;
+import com.palantir.conjure.spec.ListType;
+import com.palantir.conjure.spec.LogSafety;
+import com.palantir.conjure.spec.MapType;
+import com.palantir.conjure.spec.OptionalType;
+import com.palantir.conjure.spec.PrimitiveType;
+import com.palantir.conjure.spec.SetType;
+import com.palantir.conjure.spec.Type;
+import com.palantir.conjure.spec.TypeName;
+import java.util.Optional;
+
+public final class SafetyUtils {
+
+    private SafetyUtils() {}
+
+    public static Optional<LogSafety> getMaybeExternalSafety(AliasDefinition alias) {
+        if (alias.getAlias().accept(WrapsExternalType.INSTANCE)) {
+            return alias.getAlias().accept(GetMaybeExternalSafety.INSTANCE);
+        }
+        return alias.getSafety();
+    }
+
+    public static Optional<LogSafety> getMaybeExternalSafety(FieldDefinition field) {
+        if (field.getType().accept(WrapsExternalType.INSTANCE)) {
+            return field.getType().accept(GetMaybeExternalSafety.INSTANCE);
+        }
+        return field.getSafety();
+    }
+
+    public static Optional<LogSafety> getMaybeExternalSafety(ArgumentDefinition argument) {
+        if (argument.getType().accept(WrapsExternalType.INSTANCE)) {
+            return argument.getType().accept(GetMaybeExternalSafety.INSTANCE);
+        }
+        return argument.getSafety();
+    }
```

Comment:
I might reframe and refactor these a bit to decouple behavior from external types. There's a class of values (applies to fields, arguments, and aliases) which require their safety to be explicitly listed. So, given a `Type`, we need to determine whether the generated class declares safety itself, or if we need to add an annotation where the type is used.

Primitives (with the exception of bearer-token) and external-type-imports require safety at each place they're used. Determining the safety value can be handled using the existing `SafetyEvaluator` class.

Reply:
Ah, so I could instead think of this class as getting the usage time safety, which, for primitives, is declared in the wrapper type (field, argument, alias) itself and for external imports, is declared in the external-import-type itself.

Reply:
Yep!